packages/wallet/backend/src/gatehub/service.ts

```diff
@@ -40,22 +40,41 @@ export class GateHubService {
     // TODO: handle other webhook types
     switch (data.event_type) {
       case 'id.verification.accepted':
-        await this.markUserAsVerified(data.user_uuid)
+        const { user_uuid: gateHubUserId } = data
+
+        const user = await User.query().findOne({ gateHubUserId })
+
+        // if user is already verified (for manual verify cases)
+        // we skip the approveUserToGateway and overrideRiskLevel in addUserToGateway
+        // but still execute the function in order to store gatehub userState
+        if (user?.kycVerified) {
```

Comment:
if it is not verified we need to extract user data
if it is verified already indb, it should have the data in place

Reply:
if it is not already verified (manually), PR keeps the old flow.

It only adds the case in which user was already verified, so that it stores the data in this case as well.

Reply:
If we have issues with iframe sending messages in frontend, we will not receive the frontend call to `addUserToGateway`

However, we would manually approve it, and then receive the webhook from gatehub to verify the user.
Only in this case do we need to refetch userState from gatehub in order to sync our db.

Am I missing something?

Reply:
I feel like this call should happen if the user KYC status is false and it doesn't have a `firstName` and `lastName` in our database.

Reply:
agreed. 
Misinterpreted order of execution between webhook and addUserToGateway controller.

Added lastName condition